### DISCUSSION TITLE
How to import fastify error codes?

### DISCUSSION THREAD

**Discussion by teemupiiroinenwirepas** [fastify/fastify#3842] [Q&A]
We would like to map fastify error codes from https://github.com/fastify/fastify/blob/main/lib/errors.js to our own error codes, but we cannot find a way how to import the codes. How the importing should be done, or how the error code mapping should be done to the application specific error codes?

Br,
Teemu

**Comment by mcollina** [comment]
Currently that's not possible. However it _should_ be possible, would you like to export them as `fastify.errorCodes`?

**Reply by teemupiiroinenwirepas** [reply]
You mean that I would make a pull request about the change?

**Reply by mcollina** [reply]
Yes exactly!

**Reply by teemupiiroinenwirepas** [reply]
The way that I would do it, would be to create "enumeration" object of the error names, export that and also use it when calling "createError" in the errors.js instead of the hard coded name. I'm not sure would it mean that the same enumeration would be needed to be defined in the d.ts files as Fastify does not use TypeScript internally (I don't have experience on this).

**Reply by teemupiiroinenwirepas** [reply]
I can make that PR if you think that the way described in the previous post is feasible. I would still need some instructions how to handle the change in the TypeScript side.
